### ISSUE TITLE
"release operation in progress" should be predicated by room ID?

### ISSUE THREAD

**Issue by lawrence-forooghian** [ably/specification#251]
> *** @(CHA-RC1f3)@ @[Testable]@ If no @CHA-RC1g@ release operation is in progress, a new room instance shall be created, added to the room map and returned as the future value.
> *** @(CHA-RC1f4)@ @[Testable]@ If a @CHA-RC1g@ release operation is in progress, the entry shall be added to the room map, but the new room instance must not be created until the release operation has resolved. The future shall then subsequently resolve with the new room value.
> *** @(CHA-RC1g2)@ @[Testable]@ If the room does not exist in the room map, and no release operation is in progress, this operation is no-op.

I’m assuming (and have implemented this assumption in Swift) that all of these should be "release operation is in progress _for this room ID_"?

**Comment by sacOO7** [comment]
@lawrence-forooghian yes, your assumption is right, `release operation in progress` is only valid for the given argument `roomId`.

**Comment by sacOO7** [comment]
Also, just to add 
`CHA-RC1f3` has been interpreted as per `chat-js` implementation, which again made it difficult to understand by looking at code.
We don't immediately create a new room instance in the case of kotlin, rather wait for current release operation to finish and then create new instance and return it back.

**Comment by AndyTWF** [comment]
Yes you're correct - I have clarified in https://github.com/ably/specification/pull/255
